src/main/java/com/adobe/aem/compgenerator/AemCompGenerator.java

```diff
@@ -40,44 +72,151 @@
+            try {
+                InputStream input = AemCompGenerator.class.getResourceAsStream("/resources/" + configSamplePath);
+                if (input == null) {
+                    input = AemCompGenerator.class.getClassLoader().getResourceAsStream(configSamplePath);
+                }
+                byte[] buffer = new byte[input.available()];
+                input.read(buffer);
+
+                OutputStream outStream = new FileOutputStream(configPath);
+                outStream.write(buffer);
+            } catch (IOException e) {
+                LOG.error(e);
+                throw new GeneratorException("Could not initialize data config file");
             }
+        }
+
+        DeploymentInfo servletBuilder = deployment()
+                .setClassLoader(AemCompGenerator.class.getClassLoader())
+                .setContextPath("/api")
+                .setDefaultEncoding("UTF-8")
+                .setDeploymentName("componentgen.war")
+                .addServlet((Servlets.servlet("global", ConfigurationReadWriteServlet.class)
+                        .addMapping("/global")))
+                .addServlet((Servlets.servlet("properties", PropertyBuilderServlet.class)
+                        .addMapping("/properties")))
+                .addServlet((Servlets.servlet("child-properties", ChildPropBuilderServlet.class)
+                        .addMapping("/child-properties")))
+                .addServlet((Servlets.servlet("tabs", TabBuilderServlet.class)
+                        .addMapping("/tabs")));
 
-            String compDir = config.getProjectSettings().getAppsPath() + "/"
-                    + config.getProjectSettings().getComponentPath() + "/"
-                    + config.getType() + "/" + config.getName();
-            config.setCompDir(compDir);
+        DeploymentManager manager = defaultContainer().addDeployment(servletBuilder);
+        manager.deploy();
+        try {
+            //HttpHandler servletHandler = manager.start();
+            PathHandler servletHandler = Handlers.path(Handlers.redirect("/api"))
+                    .addPrefixPath("/api", manager.start());
+            Path staticPath = Paths.get("src/main/resources/static/build");
 
-            //builds component folder and file structure.
-            ComponentUtils generatorUtils = new ComponentUtils(config);
-            generatorUtils.buildComponent();
+            String protocol = AemCompGenerator.class.getResource("").getProtocol();
+            // if we are running the app from a Jar file... extract the static web dir
+            if (Objects.equals(protocol, "jar")) {
+                try {
+                    Path tempDirWithPrefix = Files.createTempDirectory("comp_gen_web");
+                    tempDirWithPrefix.toFile().deleteOnExit(); // cleanup temp folder when exiting app
+                    JarFile jarFile = CommonUtils.jarForClass(AemCompGenerator.class, null);
```

Comment:
Do we truly need to get a reference to the jar file for this, or can we simply get resources from the current classloader?